include/fiction/algorithms/simulation/sidb/clustercomplete.hpp

```diff
@@ -0,0 +1,476 @@
+        for (uint64_t i = 0; i < num_threads_to_use; ++i)
+        {
+            ranges.emplace_back(start, end);
+            start = end + 1;
+            end   = i == num_threads_to_use - 2 ? top_level_multisets - 1 : start + chunk_size - 1;
+        }
+
+        std::vector<std::thread> threads_vec{};
+        threads_vec.reserve(num_threads_to_use);
+
+        for (const std::pair<uint64_t, uint64_t>& range : ranges)
+        {
+            threads_vec.emplace_back(
+                [&]
+                {
+                    // iterate over all multiset charge configurations in the assigned range
+                    for (uint64_t ix = range.first; ix <= range.second; ++ix)
+                    {
+                        // iterate over all cluster charge assignments in the multiset charge configuration
+                        for (sidb_cluster_state_composition& composition :
+                             std::next(top_cluster->charge_space.cbegin(), static_cast<int64_t>(ix))->compositions)
+                        {
+                            sidb_clustering_state clustering_state{};
+                            clustering_state.reserve(charge_layout.num_cells());
+
+                            for (sidb_cluster_state& cst : composition)
+                            {
+                                clustering_state.emplace_back(std::make_unique<sidb_cluster_state>(std::move(cst)));
+                            }
+
+                            add_physically_valid_charge_confs(clustering_state);
+                        }
+                    }
+                });
+        }
+
+        for (std::thread& t : threads_vec)
+        {
+            t.join();
+        }
+    }
+
+    sidb_simulation_result<Lyt> res{};
+    std::mutex                  res_mutex{};
+
+    const charge_distribution_surface<Lyt> charge_layout;
+    const std::array<double, 4>            mu_bounds_with_error;
+    const uint64_t                         num_threads;
+};
+
+}  // namespace detail
+
+/**
+ * *ClusterComplete* is an *exact* simulator that radicalizes the SiDB simulation domain with its potential. SiDB logic
+ * simulation in *base 3* with a 100% certainty of accuracy is made available to the domain for well over 50 SiDBs. It
+ * was created by Willem Lambooy for his Master's Thesis in Theoretical Computing Science at Radboud University,
+ * Nijmegen, and employs the dual concepts of construction and destruction, ie. abstraction and computation, ie.
+ * pre-simulation and simulation in practice, breaking the rapid exponential growth in problem complexity for SiDB
+ * simulation problems in application.
+ *
+ * The key ingredient is the *Ground State Space* algorithm, which constructs a minimal search space of charge
+ * configurations that adhere to the critical population stability criterion. In particular, it generalizes Jan
+ * Drewniok's physically informed space pruning concept employed in his game-changing *QuickExact* simulator to apply to
```

Comment:
```suggestion
 * configurations that adhere to the critical population stability criterion. In particular, it generalizes the physically informed space pruning concept employed in the *QuickExact* simulator to apply to
```

Typically, we do not add names in ``fiction`` and only add the reference after successful publication. And try to avoid judgments.

Reply:
Got it. Changing this later